### PR TITLE
[Sprint: 50] XD-3016 Documentation for the Kafka message bus

### PR DESCRIPTION
**Pull request by mbogoevici**
- update message bus properties;
- clarify the behaviour of the Kafka message bus with respect to partitioning, as well as partitioned streams

### CODE REVIEW

File: src/docs/asciidoc/Application-Configuration.asciidoc [hunk 1]
```diff
@@ -310,52 +310,112 @@ will be attempted repeatedly, with no termination. Also see `republishToDLQ`
...
+NOTE: To ensure the proper functioning of the Kafka Message Bus, you must eanble log cleaning in your Kafka
+configuration.  This is set using the configuration variable `log.cleaner.enable=true`.
+See the https://cwiki.apache.org/confluence/display/KAFKA/Log+Compaction[Kafka documentation] for additional
+configuration options for log cleaning.
 
-NOTE: At this time, the Kafka message bus does not support job processing. This feature will be available in a future release.
+NOTE: At this time, the Kafka message bus does not support job processing.
+
+NOTE: The Kafka message bus does not support `count=0` for module deployments, and therefore, it does not support
+direct binding of modules. This feature will be available in a future release. In the mean time, if direct communication
```
Comment (garyrussell) [inline src/docs/asciidoc/Application-Configuration.asciidoc]: `s/mean time/meantime/`

**Comment by garyrussell** [comment]
I didn't review the properties stuff, but the partitioning descriptions LGTM.

File: src/docs/asciidoc/MessageBus.asciidoc [hunk 2]
```diff
@@ -210,3 +211,66 @@ The following query params are supported:
...
+[[kafka-bus-partitions]]
+=== Kafka Message Bus Partition Control
+
+This section describes how topic partitioning functions when using Kafka as transport.
+
+[[kafka-partition-control]]
+==== Controlling the partition count of a transport topic
+
+The _KafkaMessageBus_ will attempt to set the number of partitions in a transport topic as _consumerCount * consumerConcurrency_,
+either by creating the topic with the required number of partitions or by repartitioning it in case it exists.
```
Comment (ilayaperumalg) [inline src/docs/asciidoc/MessageBus.asciidoc]: s/it in case/if in case

File: config/servers.yml [hunk 3]
```diff
@@ -91,7 +91,6 @@
 #        concurrency:               1
 #        requiredAcks:              1
 #        compressionCodec:          none
-#        autoCommitEnabled:         true
```
Comment (ilayaperumalg) [inline config/servers.yml]: Is the reason to remove this option to imply `KafkaMessageBus`always does the auto acknowledgement of kafka messages and uses OffsetManager?

File: src/docs/asciidoc/Application-Configuration.asciidoc [hunk 4]
```diff
@@ -310,52 +310,112 @@ will be attempted repeatedly, with no termination. Also see `republishToDLQ`
...
+ between modules is necessary for Kafka deployments, xref:Modules#composing-modules[composite modules] should be used instead.
 
 [source,yaml]
 ----
   messagebus:
     kafka:
```
Comment (ilayaperumalg) [inline src/docs/asciidoc/Application-Configuration.asciidoc]: - numOfKafkaPartitionsForCountEqualsZero : I guess this property needs to be removed from `servers.yml`
- offsetStoreBatchEnabled: This property is missing in the documentation

**Comment by ilayaperumalg** [comment]
@mbogoevici I can fix the typo while merging. But I need your confirmation on the above questions.Thanks!

File: config/servers.yml (hunk 3 above)
Comment (mbogoevici) [inline config/servers.yml]: Yes,  we should allow for disabling `autoCommitEnabled` at consumer level in a future release, but with the only option available right now to set it globally, it's a very cumbersome setting for the end user. It also requires thorough documentation, including a description of the acknowledgment mechanism in Spring Integration Kafka. 

Captured as a requirement as part of https://jira.spring.io/browse/XD-3164.

File: src/docs/asciidoc/Application-Configuration.asciidoc (hunk 4 above)
Comment (mbogoevici) [inline src/docs/asciidoc/Application-Configuration.asciidoc]: Correct - the former is a slip, the second is not available as an option anymore either (0.8.2 producers are always batching) - will remove.

File: src/docs/asciidoc/MessageBus.asciidoc (hunk 2 above)
Comment (mbogoevici) [inline src/docs/asciidoc/MessageBus.asciidoc]: Actually, 'it' was referring to the topic. Probably a comma would have worked better :)

**Comment by mbogoevici** [comment]
Added a few fixes.

**Comment by ilayaperumalg** [comment]
LGTM. Squashed, rebased and merged as a4b97dcc2db1d93a863f465f527702d09cad57bc

**Pull request closed without merging**